devices/mini.go

```diff
@@ -0,0 +1,73 @@
+package devices
+
+import (
+	"image"
+
+	streamdeck "github.com/magicmonkey/go-streamdeck"
+)
+
+var (
+	miniName                     string
+	miniButtonWidth              uint
+	miniButtonHeight             uint
+	miniImageReportPayloadLength uint
+	miniImageReportHeaderLength  uint
+	miniImageReportLength        uint
+)
+
+// GetImageHeaderMini returns the USB comms header for a button image for the XL
+func GetImageHeaderMini(bytesRemaining uint, btnIndex uint, pageNumber uint) []byte {
+	var thisLength uint
+	if miniImageReportPayloadLength < bytesRemaining {
+		thisLength = miniImageReportPayloadLength
+	} else {
+		thisLength = bytesRemaining
+	}
+	header := []byte{
+		'\x02',
+		'\x01',
+		byte(pageNumber),
+		0,
+		get_header_element(thisLength, bytesRemaining),
+		byte(btnIndex + 1),
+		'\x00',
+		'\x00',
+		'\x00',
+		'\x00',
+		'\x00',
+		'\x00',
+		'\x00',
+		'\x00',
+		'\x00',
+		'\x00',
+	}
+
+	return header
+}
+
+func get_header_element(thisLength, bytesRemaining uint) byte {
+	if thisLength == bytesRemaining {
+		return '\x01'
+	} else {
+		return '\x00'
+	}
+}
+
+func init() {
+	miniName = "Streamdeck Mini"
+	miniButtonWidth = 80
+	miniButtonHeight = 80
+	miniImageReportPayloadLength = 1024
+	streamdeck.RegisterDevicetype(
+		miniName, // Name
+		image.Point{X: int(miniButtonWidth), Y: int(miniButtonHeight)}, // Width/height of a button
+		0x63,                        // USB productID
+		[]byte{0x0B, 0x63},      // Reset packet
```

Comment:
Based on the Wireshark dump and a document I found detailing the Stream Deck USB protocol, this is the Reset packet. the `\x03` packet are the String packets that make up information like device name and Serial number. I found that if I copied the device files from the XL or orig and only made a few edits, the Device reset call would actually overwrite the data in the string descriptor field. 

Messed up my Serial Number temporarily this way. The Stream Deck USB Protocol documentation combined with my findings on the Mini would indicate this is the same for all Stream Deck devices.